textattack/search_methods/PSO_algorithm.py

```diff
@@ -0,0 +1,266 @@
+                prob_list.append(0)
+                continue
+            p, w = self.gen_most_change(x, i, neighbors_list[i])
+            w_list.append(w)
+            prob_list.append(p)
+
+        prob_list = self.norm(prob_list)
+
+        h_score = prob_list
+        h_score = np.array(h_score)
+        return h_score, w_list
+
+    def norm(self, n):
+
+        tn = []
+        for i in n:
+            if i <= 0:
+                tn.append(0)
+            else:
+                tn.append(i)
+        s = np.sum(tn)
+        if s == 0:
+            for i in range(len(tn)):
+                tn[i] = 1
+            return [t / len(tn) for t in tn]
+        new_n = [t / s for t in tn]
+
+        return new_n
+
+    # for un-targeted attacking
+    def gen_most_change(self, x_cur, pos, replace_list):
+        orig_result, self.search_over = self.get_goal_results(
+            [x_cur], self.correct_output
+        )
+        if self.search_over:
+            return 0, x_cur.words[pos]
+        new_x_list = [x_cur.replace_word_at_index(pos, w) for w in replace_list]
+        # new_x_list = self.get_transformations(
+        #     x_cur,
+        #     original_text=self.original_attacked_text,
+        #     indices_to_modify=[pos],
+        # )
+        new_x_results, self.search_over = self.get_goal_results(
+            new_x_list, self.correct_output
+        )
+        new_x_scores = np.array([r.score for r in new_x_results])
+        new_x_scores = (
+            new_x_scores - orig_result[0].score
+        )  # minimize the score of ground truth
+        if len(new_x_scores):
+            return (
+                np.max(new_x_scores),
+                new_x_list[np.argsort(new_x_scores)[-1]].words[pos],
+            )
+        else:
+            return 0, x_cur.words[pos]
+
+    def _get_neighbors_list(self, attacked_text):
+        """
+        Generates this neighbors_len list
+        Args:
+            attacked_text: The original text
+        Returns:
+            A list of number of candidate neighbors for each word
+        """
+        words = attacked_text.words
+        neighbors_list = [[] for _ in range(len(words))]
+        transformations = self.get_transformations(
+            attacked_text, original_text=self.original_attacked_text
+        )
+        for transformed_text in transformations:
+            try:
+                diff_idx = attacked_text.first_word_diff_index(transformed_text)
+                neighbors_list[diff_idx].append(transformed_text.words[diff_idx])
+            except:
+                assert len(attacked_text.words) == len(transformed_text.words)
+                assert all(
+                    [
+                        w1 == w2
+                        for w1, w2 in zip(attacked_text.words, transformed_text.words)
+                    ]
+                )
+        neighbors_list = [np.array(x) for x in neighbors_list]
+        return neighbors_list
+
+    def equal(self, a, b):
```

Comment:
**private methods**
can you make the methods that will never be called from outside of this class private (prefix with `_`?) so:
- `equal -> _equal`
- `turn -> _turn`
- `sigmoid -> _sigmoid`
- `generate_population -> _generate_population`
- `gen_h_score -> _gen_h_score`